src/controllers/ManageTermController.java

```diff
@@ -3,33 +3,42 @@
 import java.net.URL;
 import java.util.ResourceBundle;
 
+import database.TermDatabaseController;
 import javafx.fxml.FXML;
 import javafx.fxml.Initializable;
 import javafx.scene.control.DatePicker;
+import javafx.scene.control.TextField;
 import javafx.stage.Stage;
 
 public class ManageTermController implements Initializable {
-	@FXML
-	DatePicker startDatePicker;
-	@FXML
-	DatePicker endDatePicker;
+	@FXML DatePicker startDatePicker;
+	@FXML DatePicker endDatePicker;
+	@FXML TextField txtFldDescription;
 	private ResourceBundle rb;
 	//Pointer to the Stage that contains the Pane
-	Stage dialogStage;
+	Stage dialogStage = null;
 	
 	@Override
 	public void initialize(URL location, ResourceBundle resources) {
 		this.rb = resources;
 	}
 	
 	
-	public void btnCancelAction(){				
+	public void btnCancelAction(){
 		dialogStage = (Stage) startDatePicker.getScene().getWindow();
 		dialogStage.close();
 	}
 	
 	public void btnAcceptAction(){
-		//TODO add functionality with connection to DB
+		try{
+			TermDatabaseController.insertNewTerm(startDatePicker.getValue().toString(), 
+					endDatePicker.getValue().toString(), txtFldDescription.getText());
+			dialogStage = (Stage) startDatePicker.getScene().getWindow();
+			dialogStage.close();
+			
+		}catch(Exception e){
```

Comment:
why try? 🐋

Reply:
To catch the exception in case the user clicks on "accept" and the dialog fields are empty or some are missing. Should I remove it?

Reply:
😐